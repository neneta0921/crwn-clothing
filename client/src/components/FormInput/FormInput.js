import {
  FormInputContainer,
  FormInputLabel,
  GroupContainer
} from './FormInputStyles'

export const FormInput = ({ handleChange, label, ...otherProps }) => (
  <GroupContainer>
    <FormInputContainer onChange={handleChange} {...otherProps} />
    {label
      ? (<FormInputLabel>
          { label }
        </FormInputLabel>)
      : null
    }
  </GroupContainer>
)