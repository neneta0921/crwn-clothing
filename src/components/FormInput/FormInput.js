import {
  FormInputContainer,
  FormInputLabel,
  GroupContainer
} from './FormInput.styles'

export const FormInput = ({ handleChange, label, ...otherProps }) => (
  <GroupContainer>
    <FormInputContainer onChange={handleChange} {...otherProps} />
    {label
      ? (<FormInputLabel>
          { label }
        </FormInputLabel>)
      : null
    }
  </GroupContainer>
)