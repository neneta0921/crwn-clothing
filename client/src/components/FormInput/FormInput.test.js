import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { FormInput } from './FormInput';

describe('FormInput component', () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();
    const mockProps = {
      label: 'email',
      value: 'test@gmail.com',
      handleChange: mockHandleChange,
    };

    wrapper = shallow(<FormInput {...mockProps} />);
  });

  it('should render FormInput component', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call handleChange method when input changes', () => {
    wrapper.find('FormInputContainer').simulate('change');
    expect(mockHandleChange).toHaveBeenCalled();
  });

  it('should render FormInputLabel if there is a label', () => {
    expect(wrapper.exists('FormInputLabel')).toBe(true);
  });

  it('should not render FormInputLabel if there is no label', () => {
    const mockNewProps = {
      label: '',
      value: 'test@gmail.com',
      handleChange: mockHandleChange,
    };

    const newWrapper = shallow(<FormInput {...mockNewProps} />);
    expect(newWrapper.exists('FormInputLabel')).toBe(false);
  });
});
