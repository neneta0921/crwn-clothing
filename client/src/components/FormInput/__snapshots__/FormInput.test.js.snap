// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render FormInput component 1`] = `
<div
  className="sc-bdfBwQ ickNGW"
>
  <input
    className="sc-gsTCUz gpJyyF"
    onChange={[MockFunction]}
    value="test@gmail.com"
  />
  <label
    className="sc-dlfnbm hePAmY"
  >
    email
  </label>
</div>
`;
